src/EFCore.PG/Storage/Internal/Mapping/NpgsqlArrayArrayTypeMapping.cs

```diff
@@ -101,6 +95,14 @@ static bool Compare(TElem[] a, TElem[] b, ValueComparer<TElem> elementComparer)
                 return true;
             }
 
+            static int GetHashCode(TElem[] source, ValueComparer<TElem> elementComparer)
+            {
+                var hash = new HashCode();
+                foreach (var el in source)
+                    hash.Add(elementComparer.GetHashCode(el));
```

Comment:
```suggestion
                foreach (var el in source)
                    hash.Add(el, elementComparer);
```

Reply:
This is more complicated for nullable, since the ValueComparer is over the non-nullable type.

But you're right that it's more correct (hash.Add accepts a value and calls GetHashCode on it, rather than the hashcode). I will make it work.